subsys/net/ip/tcp2.c

```diff
@@ -1054,18 +1054,26 @@ static void tcp_timewait_timeout(struct k_work *work)
 {
 	struct tcp *conn = CONTAINER_OF(work, struct tcp, timewait_timer);
 
+	k_mutex_lock(&tcp_lock, K_FOREVER);
+
 	NET_DBG("conn: %p %s", conn, log_strdup(tcp_conn_state(conn, NULL)));
 
 	/* Extra unref from net_tcp_put() */
 	net_context_unref(conn->context);
+
+	k_mutex_unlock(&tcp_lock);
 }
 
 static void tcp_establish_timeout(struct tcp *conn)
 {
+	k_mutex_lock(&tcp_lock, K_FOREVER);
+
 	NET_DBG("Did not receive %s in %dms", "ACK", ACK_TIMEOUT_MS);
 	NET_DBG("conn: %p %s", conn, log_strdup(tcp_conn_state(conn, NULL)));
 
 	(void)tcp_conn_unref(conn);
+
+	k_mutex_unlock(&tcp_lock);
```

Comment:
It looks like `tcp_conn_unref()` already locks `&tcp_lock`, so why do it here as well?

Reply:
You have a point, I seemed to put these too eagerly here. In fact this commit does not make much sense actually. I will remove it.